woocommerce/woocommerce-gutenberg-products-block
# Product Gallery: Add a better default Dialog template part

This PR tries to improve upon the default Dialog template.

## What

This PR aims to address some of the concerns of woocommerce/woocommerce#42217.

- This PR **does not** answer the question of how should Thumbnails be displayed (especially when there's a lot of them).
- This PR **does not** answer the question of how should the Next/Previous buttons be displayed in the Dialog.

## Why

This PR changes how the Large Image behaves (and gets re-sized) in the Dialog in relation to the Thumbnails.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. With a block theme, open `Appearance > Editor`.
2. Open `Template Parts > Product Gallery`.
3. If you had one customized earlier, reset it to load the new default template.
4. Change the `Row` alignment to `Top` and save (this won't be needed when woocommerce/woocommerce-blocks#11104 is fixed):
<img width="844" alt="Product_Gallery_‹_Template_Part_‹_Editor_‹_newproductgallery_—_WordPress_i_Messenger" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/54be246d-6f87-40e9-be12-2de5bbad74a7">

5. Open a product page on the frontend **with multiple product images** and select the Large Image.
6. Make sure the Dialog opens and resizes the Large Image correctly (without overflowing the Dialog).


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|<img width="1350" alt="V-Neck_T-Shirt_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/9963a9d8-99e1-4857-b6d5-4dfdc4252b6e">|<img width="1350" alt="V-Neck_T-Shirt_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/9519fcd2-2d62-4eac-b17d-03bff10bd9c6">|

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Gallery: Update the default dialog template part.
